### PR TITLE
Docs about properties updates and some guidelines for migration

### PR DESCRIPTION
**Pull request by czoido**
Docs for https://github.com/conan-io/conan/pull/10098 and https://github.com/conan-io/conan/pull/10099

### CODE REVIEW

File: reference/conanfile/attributes.rst [hunk 1]
```diff
@@ -1052,11 +1058,10 @@ recommended way of setting this information is using ``set_property`` and ``get_
...
 ``cmake_module_file_name``, ``pkg_config_name`` and ``cmake_build_modules`` are defined to allow
 migrating ``names``, ``filenames`` and ``build_modules`` properties to this model. In Conan 2.0 this
-will be the default way of setting these properties and also passing custom properties to generators.
-These properties are supported by the new :ref:`CMakeDeps<CMakeDeps Properties>` and
-:ref:`PkgConfigDeps<PkgConfigDeps Properties>` generators.
+will be the default way of setting these properties for all generators and also passing
+custom properties to generators.
 
-New properties defined for *CMake* generators family:
+New properties defined for for *CMake* generators family, used by :ref:`CMakeDeps<CMakeDeps>` generator:
```
Comment (czoido) [inline reference/conanfile/attributes.rst:1064]:
```suggestion
New properties defined for *CMake* generators family, used by :ref:`CMakeDeps<CMakeDeps>` generator:
```

File: reference/conanfile/attributes.rst [hunk 2]
```diff
@@ -1081,84 +1090,174 @@ For most cases, it is recommended not to use the ``generator`` argument. The pro
...
+**Migrating from the .names and .filenames model to set_property()**
+
+Migrating from `.names` to ``set_property()`` is easy, but there are some details to take
+into account for properties like ``cmake_target_name`` and ``cmake_file_name``. Let's see
+some examples.
+
+**Migrating from .names to cmake_target_name**
+
+It is important to note that ``cmake_target_name`` is **not** goint to take the same value
+as the ``.names`` attribute did. With the ``.names`` attribute if you set a name for the
```
Comment (czoido) [inline reference/conanfile/attributes.rst:1115]:
```suggestion
as the ``.names`` attribute did. With the ``.names`` attribute, if you set a name for the
```

File: reference/conanfile/attributes.rst [hunk 3]
```diff
@@ -1081,84 +1090,174 @@ For most cases, it is recommended not to use the ``generator`` argument. The pro
...
+
+    def package_info(self):
+        ...
+        self.cpp_info.components.names["cmake_find_package"] = "myname"
+        ...
+
+Will create a CMake target named ``myname::myname``.
+
+The property ``cmake_target_name`` accepts **complete** target names. That means that the
+name you set with this property will the the one that is added to the CMake generated
```
Comment (czoido) [inline reference/conanfile/attributes.rst:1129]:
```suggestion
name you set with this property will be the one that is added to the CMake generated
```

**Review by memsharded — APPROVED** [review]

File: reference/conanfile/attributes.rst [hunk 4]
```diff
@@ -1038,11 +1038,17 @@ has **the same default directories**.
...
 
-Using ``names``, ``filenames`` and ``build_modules`` is **deprecated** from Conan 1.42. The
-recommended way of setting this information is using ``set_property`` and ``get_property`` methods
-(available since Conan 1.36) of the ``cpp_info`` object.
+.. warning::
+
+    Using ``set_property`` and ``get_property`` methods for ``cpp_info`` is an **experimental**
+    feature subject to breaking changes in future releases.
+
+Using ``names``, ``filenames`` and ``build_modules`` will not work any more for new
```
Comment (lasote) [inline reference/conanfile/attributes.rst:1048]:
```suggestion
Using ``.names``, ``.filenames`` and ``.build_modules`` will not work any more for new
```

File: reference/conanfile/attributes.rst [hunk 5]
```diff
@@ -1081,84 +1090,176 @@ For most cases, it is recommended not to use the ``generator`` argument. The pro
...
+
+**Migrating from the .names and .filenames model to set_property()**
+
+Migrating from `.names` to ``set_property()`` is easy, but there are some details to take
+into account for properties like ``cmake_target_name`` and ``cmake_file_name``. Let's see
+some examples.
+
+**Migrating from .names to cmake_target_name**
+
+It is important to note that ``cmake_target_name`` is **not** goint to take the same value
```
Comment (lasote) [inline reference/conanfile/attributes.rst:1114]:
```suggestion
It is important to note that ``cmake_target_name`` is **not** going to take the same value
```

File: reference/conanfile/attributes.rst (hunk 4 above)
Comment (lasote) [inline reference/conanfile/attributes.rst:1048]:
```suggestion
Using ``.names``, ``.filenames`` and ``.build_modules`` will not work any more for new
```

**Review by lasote — CHANGES_REQUESTED** [review]
I would try to move some content to the CMakeDeps page and others to the migration page.

**Comment by czoido** [comment]
This section has grown too much, I will try to reorganize these contents, maybe moving parts to the migrations section.

File: conan_v2.rst [hunk 6]
```diff
@@ -1,32 +1,313 @@
...
+name you set with this property will be the one added to the CMake generated
+files without appending any more information to it. 
+To translate the last example to the set_property model you should add the following
+declaration:
+
+.. code-block:: python
+
+    def package_info(self):
+        ...
+        self.cpp_info.components.names["cmake_find_package"] = "myname::myname"
```
Comment (memsharded) [inline conan_v2.rst:135]: Legacy way?

**Review by memsharded — APPROVED** [review]
It is good, but I would move the ``.. _conanv2_properties_model:`` section to its own page under "conan2migration" subfolder or something. Too heavy for something that many users at companies do not care about (mostly for ConanCenter)

**Review by memsharded — APPROVED** [review]

**Review by lasote — APPROVED** [review]

**Pull request merged**